### PR TITLE
Group_id and groups type

### LINKED ISSUE #57: Group_id and groups type
The value of "group_id" is used multiple times. However, it is semantically not the same as a "product_id". Following the DRY principle and to prepare for a unique product identifier where also the definition of these types will differ the corresponding definitions should be extracted to types.

### PR DESCRIPTION
**Pull request by tschmidtb51**
- resolves #57 
- add "group_id_t" and "groups_t"
- refactor to use new types
- add generic keywords for documentation

### CODE REVIEW

**Review by santosomar — APPROVED** [review]

**Review by sthagen — APPROVED** [review]

**Review by tolim — APPROVED** [review]

File: csaf_2.0/json_schema/csaf_json_schema.json [hunk 1]
```diff
@@ -107,6 +107,22 @@
...
+    "groups_t": {
+      "title": "List of group_ids",
+      "description": "Specifies a list of group_ids to give context to the parent item.",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/group_id_t"
+      },
+      "uniqueItems": true
+    },
+    "group_id_t": {
```
[the remark this answers is not in the document]
Comment (santosomar) [reply csaf_2.0/json_schema/csaf_json_schema.json:119]: During the June 24, 2020 monthly meeting, it was suggested and discussed to rename to `product_group_id_t` to make relationship to products explicit, as stated by @tolim above. The TC decided that this is the right approached (i.e., rename to `product_group_id_t`).

**Review by santosomar — CHANGES_REQUESTED** [review]
During the June 24, 2020 monthly meeting, it was suggested and discussed to rename to product_group_id_t to make relationship to products explicit, as stated by @tolim above. The TC decided that this is the right approached (i.e., rename to product_group_id_t).

File: csaf_2.0/json_schema/csaf_json_schema.json [hunk 2]
```diff
@@ -107,6 +107,22 @@
         }
       }
     },
+    "groups_t": {
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/json_schema/csaf_json_schema.json]: Addressed in [https://github.com/oasis-tcs/csaf/pull/58/commits/2a61d238696fd2747ea678081a7fedf4a4010830](https://github.com/oasis-tcs/csaf/pull/58/commits/2a61d238696fd2747ea678081a7fedf4a4010830).

File: csaf_2.0/json_schema/csaf_json_schema.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/json_schema/csaf_json_schema.json]: Addressed in [https://github.com/oasis-tcs/csaf/pull/58/commits/2a61d238696fd2747ea678081a7fedf4a4010830](https://github.com/oasis-tcs/csaf/pull/58/commits/2a61d238696fd2747ea678081a7fedf4a4010830)

**Review by santosomar — APPROVED** [review]
Thank you for taking care of this so quickly.

File: csaf_2.0/json_schema/csaf_json_schema.json [hunk 3]
```diff
@@ -708,7 +723,7 @@
                   }
                 },
                 "group_ids": {
-                  "$ref": "#/definitions/products_t"
+                  "$ref": "#/definitions/product_groups_t"
```
Comment (sthagen) [inline csaf_2.0/json_schema/csaf_json_schema.json:726]: :+1:

**Pull request merged**